### PR TITLE
Implement support for Pydantic models

### PR DESCRIPTION
**Pull request by NoelTautges**
This pull request implements support for [Pydantic](https://pydantic-docs.helpmanual.io) models, including tests.

Coverage:
```
----------- coverage: platform win32, python 3.9.5-final-0 -----------
Name                      Stmts   Miss  Cover   Missing
-------------------------------------------------------
itemadapter\__init__.py       3      0   100%
itemadapter\adapter.py      158      0   100%
itemadapter\utils.py         69      0   100%
-------------------------------------------------------
TOTAL                       230      0   100%
```

If this is better suited for custom registration in my own projects than a pull request, I totally understand! I just saw requested it once already (#42) and thought that since it was a theme, a pull request might be useful.

### CODE REVIEW

**Comment by elacuesta** [comment]
Hey, thank you very much for the effort here. Code looks good to me, I just wonder whether or not we should add it to the main package, from a maintainability perspective. I'm +0.5 on merging, let's see what other maintainers think.

**Review by wRAR — APPROVED** [review]

**Comment by elacuesta** [comment]
@NoelTautges Seems like field metadata could be supported via the [`Field` function](https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation), could you maybe add support for it?

**Comment by NoelTautges** [comment]
@elacuesta Sure thing! Should I add every set one except for `default_factory`?

**Comment by Gallaecio** [comment]
@NoelTautges I think it should be included (if defined). If you wonder about excluding it because callables are not serializables, I don’t think field metadata needs to be serializable. It’s not included when serializing an item, if I recall correctly.

**Comment by NoelTautges** [comment]
@Gallaecio D'oh, you're totally right. I'll get on it. 👍

**Comment by wRAR** [comment]
> itemadapter/utils.py:44: error: "type" has no attribute "__fields__"  [attr-defined]

**Comment by NoelTautges** [comment]
@wRAR Thank you for your patience! A local tox run says I fixed the issue.

**Comment by NoelTautges** [comment]
Would someone mind taking another look at this pull request? I fixed the typing issue.

File: itemadapter/utils.py [hunk 1]
```diff
@@ -101,7 +145,12 @@ def get_field_meta_from_class(item_class: type, field_name: str) -> MappingProxy
             return fields_dict(item_class)[field_name].metadata  # type: ignore
         except KeyError:
             raise KeyError("%s does not support field: %s" % (item_class.__name__, field_name))
-    elif issubclass(item_class, dict):
+    elif _is_pydantic_model(item_class):
+        try:
+            return _get_pydantic_model_metadata(item_class, field_name)
+        except KeyError:
+            raise KeyError("%s does not support field: %s" % (item_class.__name__, field_name))
+    elif issubclass(item_class, dict) or _is_pydantic_model(item_class):
```
Comment (Gallaecio) [inline itemadapter/utils.py:153]:
```suggestion
    elif issubclass(item_class, dict):
```

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Looks good to me, great work!

**Comment by NoelTautges** [comment]
@Gallaecio Thank you for merging, and thank you for fixing that unreachable condition! Take care :)

**Comment by elacuesta** [comment]
Released as [`v0.3.0`](https://github.com/scrapy/itemadapter/releases/tag/v0.3.0). Thanks for the great work!

**Comment by NoelTautges** [comment]
@elacuesta Thank you :D

**Pull request merged**